cmd/av/stack_diff.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/utils/colors"
+	"github.com/spf13/cobra"
+)
+
+var stackDiffCmd = &cobra.Command{
+	Use:          "diff",
+	Short:        "generate diff between working tree and current index",
+	Long: strings.TrimSpace(`
+Generates the diff between the working tree and the current index (i.e., the diff containing all unstaged changes).
+`),
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+
+		diff, err := repo.Diff(&git.DiffOpts{})
```

Comment:
This needs an argument for `Commit` otherwise we'll only ever generate the diff between the previous commit and the working tree (so there's no reason to run `av stack diff` versus `git diff`).

We need to determine that commit by looking at the current branch, finding the parent (if any -- the branch might be a stack root), and comparing against that.

There's a bit of a wrinkle about what happens if the branch isn't up to date with its parent (i.e., the user needs to run `av stack sync`), but we can deal with that later.